### PR TITLE
adding new random quote service and a request handler

### PR DESCRIPTION
**Pull request by luis8alopez**
## Summary

- Implemented new service to retrieve a random quote from external API
- Request handler created

## Trello Card

https://trello.com/c/TT6mdZwn/32-servicios-promises-witter

### CODE REVIEW

File: app/controllers/weets.js [hunk 1]
```diff
@@ -0,0 +1,6 @@
+const weetService = require('../services/weets');
+
+exports.getQuote = async (_, res) => {
+    const data = await weetService.getRandomQuote();
```
Comment (shernanm) [inline app/controllers/weets.js:4]: Remember to use the informational logs, this will make it much easier to debug the API.

File: app/routes.js [hunk 2]
```diff
@@ -1,8 +1,11 @@
 // const controller = require('./controllers/controller');
 const { healthCheck } = require('./controllers/healthCheck');
+const { requestHandler } = require('./helpers/requestHandler');
+const { getQuote } = require('./controllers/weets');
 
 exports.init = app => {
   app.get('/health', healthCheck);
+  app.get('/weets', requestHandler(getQuote));
   // app.get('/endpoint/get/path', [], controller.methodGET);
```
Comment (shernanm) [inline app/routes.js]: These example routes can be removed.

File: app/services/weets.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+exports.getRandomQuote = async () => {
+    try {
+        const response = await axios.get(clientAPI, {
+            params: {
+                limit: limit
+            }
+        });
+        const { data } = response.data;
+        return data[0];
```
Comment (shernanm) [inline app/services/weets.js]: Here you can return all the elements of the array and you can also try to do a destructuring of the API response.

File: app/helpers/requestHandler.js [hunk 4]
```diff
@@ -0,0 +1,7 @@
+exports.requestHandler = handler => async (req, res, next) => {
+    try {
+        await handler(req, res, next)
+    } catch (error) {
+        next(error)
+    }
+}
```
Comment (shernanm) [inline app/helpers/requestHandler.js]: I don't see much sense in this handler, you could call the controller directly.

File: app/services/weets.js [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+    } catch (error) {
+        logger.error(error);
+        throw Error(error);
```
Comment (shernanm) [inline app/services/weets.js]: Usually we do not return the same error that the API returns, here you can create a custom error, you can take a look at the errors.js file.

File: app/controllers/weets.js [hunk 6]
```diff
@@ -0,0 +1,6 @@
...
+    res.status(200).send(data)
+};
```
Comment (dbenavidesv) [inline app/controllers/weets.js]: We don't actually need a REST endpoint for this feature, so this controller is not needed

**Review by shernanm — CHANGES_REQUESTED** [review]

File: app/controllers/weets.js (hunk 6 above)
Comment (shernanm) [reply app/controllers/weets.js]: For future endpoints remember to use the informational logs, this makes API debugging much easier.

File: config/development.js [hunk 7]
```diff
@@ -3,7 +3,9 @@ exports.config = {
   common: {
     database: {
       name: process.env.DB_NAME_DEV
-    }
+    },
+    clientAPI: 'https://quote-garden.herokuapp.com/api/v3/quotes',
```
Comment (dbenavidesv) [inline config/development.js]: This should be added as an Environment Variable. Putting this type of information directly in the repository can be very problematic and prone to security issues. You can check how to handle this in the `config/index.file`

File: config/development.js [hunk 8]
```diff
@@ -3,7 +3,9 @@ exports.config = {
...
+    limit: 1
```
Comment (dbenavidesv) [inline config/development.js]: Also, you should group up this properties under something like:
```
  quotesAPI: {
    url: (...),
    limit: 1
  }
```

File: config/production.js [hunk 9]
```diff
@@ -3,7 +3,9 @@ exports.config = {
   common: {
     database: {
       name: process.env.DB_NAME
-    }
+    },
+    clientAPI: 'https://quote-garden.herokuapp.com/api/v3/quotes',
+    limit: 1
```
Comment (dbenavidesv) [inline config/production.js]: A per the previous comments, this should be removed

File: app/services/weets.js [hunk 10]
```diff
@@ -0,0 +1,18 @@
...
+    }
+}
```
Comment (dbenavidesv) [inline app/services/weets.js]: This is a minor thing, but remember to always leave an empty newline in all files

**Review by luis8alopez — COMMENTED** [review]
Changes reviewed and corrections applied to the project

File: app/middlewares/errors.js [hunk 11]
```diff
@@ -5,7 +5,8 @@ const DEFAULT_STATUS_CODE = 500;
 
 const statusCodes = {
   [errors.DATABASE_ERROR]: 503,
-  [errors.DEFAULT_ERROR]: 500
+  [errors.DEFAULT_ERROR]: 500,
+  [errors.API_ERROR]: 404
```
Comment (dbenavidesv) [inline app/middlewares/errors.js]: I don't think `404` is the right one here, maybe find a better http code for unhandled external api errors

File: config/development.js [hunk 12]
```diff
@@ -3,7 +3,8 @@ exports.config = {
   common: {
     database: {
       name: process.env.DB_NAME_DEV
-    }
+    },
+
```
Comment (dbenavidesv) [inline config/development.js]: Remove this empty line

File: app/middlewares/errors.js [hunk 13]
```diff
@@ -5,7 +5,8 @@ const DEFAULT_STATUS_CODE = 500;
 
 const statusCodes = {
   [errors.DATABASE_ERROR]: 503,
-  [errors.DEFAULT_ERROR]: 500
+  [errors.DEFAULT_ERROR]: 500,
+  [errors.API_ERROR]: 500
```
Comment (dbenavidesv) [inline app/middlewares/errors.js:9]: Look up `503` it should work for this type of error as well, but `500` is fine

**Review by dbenavidesv — APPROVED** [review]

**Review by shernanm — APPROVED** [review]

**Pull request merged**